README.md

```diff
@@ -7,42 +7,87 @@
 
 
 ## About
-This is a simple REST client for C++. It wraps libcurl for HTTP requests.
+This is a simple REST client for C++. It wraps [libcurl][] for HTTP requests.
 
 ## Usage
-I tried to keep usage close to the [ruby rest-client][]. So the basic usage is:
+restclient-cpp provides two ways of interacting with REST endpoints. There is
+a simple one, which doesn't need you to configure an object to interact with
+an API. However the simple way doesn't provide a lot of configuration options
+either. So if you need more than just a simple HTTP call, you will probably
+want to check out the advanced usage.
 
-```cpp
-RestClient::method(url, content-type, params);
-// or
-RestClient::method(url, content-type, params, headers);
-```
-
-Examples:
+### Simple API
+The simple API is just some static methods modeled after the most common HTTP
+verbs:
 
 ```cpp
 #include "restclient-cpp/restclient.h"
 
+// initialize RestClient
+RestClient::init()
+
 RestClient::response r = RestClient::get("http://url.com")
 RestClient::response r = RestClient::post("http://url.com/post", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::put("http://url.com/put", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::del("http://url.com/delete")
 
-// add some headers
+// deinit RestClient. After calling this you have to call RestClient::init()
+// again before you can use it
+RestClient::disable();
+```
+
+The response is of type [RestClient::response][restclient_response] and has
+three attributes:
+
+```cpp
+RestClient::response.code // HTTP response code
+RestClient::response.body // HTTP response body
+RestClient::response.headers // HTTP response headers
+```
+
+### Advanced Usage
+However if you want more sophisticated features like connection reuse,
+timeouts or authentication, there is also a different, more configurable way.
 
+```cpp
+#include "restclient-cpp/restclient.h"
+
+// initialize RestClient
+RestClient::init();
```

Comment:
I'll have to investigate this further, but this is what libcurl says about the global functions:

> It isn't actually required that the functions be called at the beginning and end of the program -- that's just usually the easiest way to do it. It is required that the functions be called when no other thread in the program is running.

And if I want to retain the simple declaration that it's safe to use the connection object from within threads, then I can't do the `call_once` in there as I can't guarantee there aren't any other threads running. I think for the simple API this is fair and makes it a lot nicer to use.